supervisor/shared/stack.c

```diff
@@ -50,6 +53,17 @@ void allocate_stack(void) {
     } else {
         current_stack_size = next_stack_size;
     }
+    *stack_alloc->ptr = STACK_CANARY_VALUE;
+}
+
+inline bool stack_ok(void) {
+    return *stack_alloc->ptr == STACK_CANARY_VALUE;
+}
+
+inline void assert_heap_ok(void) {
+    if (!stack_ok()) {
+        mp_raise_RuntimeError(translate("Stack clobbered heap."));
```

Comment:
Since this is a very bad error, we should probably also say something like "Please restart" or "Please type ctrl-D". The heap may be damaged and may not be safe to do anything else

Reply:
Think I should make it safe mode instead? It'd work more reliably because then it wouldn't try to use the heap at all.